lib/membrane_h264_ffmpeg/decoder.ex

```diff
@@ -78,10 +94,14 @@ defmodule Membrane.H264.FFmpeg.Decoder do
     {:ok, %{state | decoder_ref: nil}}
   end
 
-  defp wrap_frames([]), do: []
+  defp wrap_frames([], []), do: []
 
-  defp wrap_frames(frames) do
-    frames |> Enum.map(fn frame -> %Buffer{payload: frame} end) ~> [buffer: {:output, &1}]
+  defp wrap_frames(best_effort_pts_list, frames) do
+    Enum.zip(best_effort_pts_list, frames)
+    |> Enum.map(fn {best_effort_pts, frame} ->
+      %Buffer{metadata: %{pts: best_effort_pts}, payload: frame}
+    end)
+    ~> [buffer: {:output, &1}]
```

Comment:
I think this might be an opportunity to get rid of bunch an use Elixir's 1.12 `then`

Reply:
bump